src/Geocodio.php

```diff
@@ -60,138 +80,253 @@ public function setApiVersion(string $apiVersion): self
+     *
+     * @see https://www.geocod.io/docs/#delete-a-list
+     */
+    public function deleteList(int $listId): mixed
     {
-        if (is_array($query)) {
-            $queryParameters += $query;
+        $response = $this->sendRequest(
+            'DELETE',
+            "lists/{$listId}",
+        );
+
+        return $this->toResponse($response);
+    }
+
+    /**
+     * Reverse geocode a coordinate or a list of coordinates
+     *
+     * @see https://www.geocod.io/docs/#reverse-geocoding
+     *
+     * @param  string|array  $query
+     * @return array|object
+     */
+    public function reverse(
+        $query,
+        array $fields = [],
+        ?int $limit = null,
+        ?string $format = null
+    ): mixed {
+        $options = [
+            RequestOptions::QUERY => [
+                'q' => $this->formattedReverseQuery($query),
+                'fields' => implode(',', $fields),
+                'limit' => $limit,
+                'format' => $format,
+            ],
+        ];
+
+        if (is_string($query) || (is_array($query) && is_numeric($query[0]))) {
+            $response = $this->sendRequest('GET', 'reverse', $options);
         } else {
-            $queryParameters['q'] = $query;
+            $options[RequestOptions::JSON] = array_map(fn ($q) => $this->formattedReverseQuery($q), $query);
+
+            $response = $this->sendRequest('POST', 'reverse', $options);
+
         }
 
-        return $this->client->get($url, [
-            'query' => $queryParameters,
-            'headers' => $this->getHeaders(),
-        ]);
-    }
+        return $this->toResponse($response);
 
-    private function performBatchRequest(string $url, array $queries, array $queryParameters): \Psr\Http\Message\ResponseInterface
-    {
-        return $this->client->post($url, [
-            'query' => $queryParameters,
-            'json' => $queries,
-            'headers' => $this->getHeaders(),
-        ]);
     }
 
-    private function handleException(\Throwable $e): void
+    protected function isSingleQuery($query): bool
     {
-        $response = $e instanceof RequestException && $e->hasResponse() ? $e->getResponse() : null;
-
-        $errorMessage = 'Error';
-        $errorCode = 0;
+        if (is_array($query)) {
+            $addressComponentKeys = array_intersect(array_keys($query), self::ADDRESS_COMPONENT_PARAMETERS);
 
-        if ($response instanceof \Psr\Http\Message\ResponseInterface) {
-            $json = @json_decode((string) $response->getBody());
+            return count($addressComponentKeys) >= 1;
+        }
 
-            if ($json && isset($json->error)) {
-                $errorMessage = $json->error;
-            }
+        return true;
+    }
 
-            $errorCode = $response->getStatusCode();
+    protected function uploadMultipartFile(
+        string $fileContents,
+        GeocodeDirection $direction,
+        string $format,
+        string $callbackWebhook,
```

Comment:
Should this be `?string $callbackWebhook = null` instead. Since it's optional?

Reply:
Good call!